yapapi/payload/vm.py

```diff
@@ -87,9 +93,45 @@ async def decorate_demand(self, demand: DemandBuilder):
         demand.add(VmRequest(package_url=image_url, package_format=VmPackageFormat.GVMKIT_SQUASH))
 
 
+@dataclass
+class _VmManifestPackage(Package):
+    manifest: str
+    manifest_sig: str
+    constraints: _VmConstraints
+
+    async def resolve_url(self) -> str:
+        return ""
+
+    async def decorate_demand(self, demand: DemandBuilder):
+        demand.ensure(str(self.constraints))
+        demand.add(VmManifestRequest(
+            manifest=self.manifest,
+            manifest_sig=self.manifest_sig,
+            package_format=VmPackageFormat.GVMKIT_SQUASH
+        ))
+
+
+async def manifest(
```

Comment:
Is there any reason for this to be an `async` function?

We've already decided that `repo` should not be `async` (and we plan to change it), so maybe we can have non-async `manifest` from the beginning.

@shadeofblue